dcm4che-tool/dcm4che-tool-dcmdir/src/main/java/org/dcm4che3/tool/dcmdir/DcmDir.java

```diff
@@ -230,7 +269,48 @@ public static void main(String[] args) {
         }
     }
 
-    public void compact(File f, File bak) throws IOException {
+    private int readCSVFile(CommandLine cl, DcmDir main, int num) throws Exception {
+        if (!cl.hasOption("record-config"))
+            throw new MissingOptionException("Missing record-config option.");
+        String delim = cl.hasOption("csv-delim") ? cl.getOptionValue("csv-delim") : ",";
+        String quote = cl.hasOption("csv-quote") && !cl.getOptionValue("csv-quote").equals("")
+                        ? cl.getOptionValue("csv-quote") : "\"";
+        BufferedReader br = null;
+        try {
+            br = new BufferedReader(new FileReader(cl.getOptionValue("csv")));
+            loadDefaultConfiguration(cl.getOptionValue("record-config"));
+            String headerline = br.readLine();
+            String[] headers = headerline.split(delim);
+            List<Integer> tags = new ArrayList<Integer>();
+            List<VR> vrs = new ArrayList<VR>();
+            for (int i = 0; i < headers.length; i++) {
+                String header = headers[i].replaceAll(quote, "");
+                int tag = DICT.tagForKeyword(header);
+                tags.add(tag);
+                vrs.add(DICT.vrOf(tag));
+            }
+
+            String nextLine;
+            while((nextLine = br.readLine()) != null) {
+                String escq = new StringBuilder().append("\\").append(quote).toString();
+                String regex = delim + "(?=(?:[^" + escq + "]*" + escq + "[^" + escq + "]*" + escq + ")*[^" + escq + "]*$)";
+                String[] values = nextLine.split(regex, -1);
```

Comment:
Performance: Better use Pattern.compile(..).split(...) and do the Pattern.compile(...) outside of the loop.